src/support.cpp

```diff
@@ -48,6 +48,261 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    unsigned int gradual_support_step_layer_count = round_divide(gradual_support_step_height, storage.getSettingInMicrons("layer_height")); // The difference in layer count between consecutive density infill areas
+
+    // make gradual_support_step_height divisable by layer_skip_count
+    float n_skip_steps_per_gradual_step = std::max(1.0f, std::ceil(gradual_support_step_layer_count / layer_skip_count)); // only decrease layer_skip_count to make it a divisor of gradual_support_step_layer_count
+    layer_skip_count = gradual_support_step_layer_count / n_skip_steps_per_gradual_step;
+
+    size_t min_layer = 0;
+    size_t max_layer = total_layer_count - 1;
+
+    const ExtruderTrain& infill_extr = *storage.meshgroup->getExtruderTrain(storage.getSettingAsIndex("support_infill_extruder_nr"));
+    const coord_t support_line_width = infill_extr.getSettingInMicrons("support_line_width");
+    const int wall_line_count = 1;  // the wall line count is used for calculating insets, and we generate support infill patterns within the insets
+
+    for (unsigned int layer_nr = 0; layer_nr < total_layer_count - 1; ++layer_nr)
+    {
+        assert(storage.support.supportLayers[layer_nr].support_infill_parts.empty() && "support infill part list is supposed to be uninitialized");
+
+        // this is the complete support areas on this layer
+        const Polygons& whole_support_areas = storage.support.supportLayers[layer_nr].supportAreas;
+
+        if (whole_support_areas.size() == 0 || layer_nr < min_layer || layer_nr > max_layer)
+        {
+            // initialize support_infill_parts empty
+            storage.support.supportLayers[layer_nr].support_infill_parts.clear();
+            continue;
+        }
+
+        // generate separate support islands and calculate density areas for each island
+        std::vector<PolygonsPart> support_islands = whole_support_areas.splitIntoParts();
+        for (unsigned int i = 0; i < support_islands.size(); ++i)
+        {
+            storage.support.supportLayers[layer_nr].support_infill_parts.emplace_back();
+            SupportInfillPart& support_infill_part = storage.support.supportLayers[layer_nr].support_infill_parts.back();
+            support_infill_part.gradual_infill_areas_per_combine_per_density.clear();
+            support_infill_part.outline = support_islands[i];
+            support_infill_part.insets.clear();
+            // generate insets and use the first inset as the infill area
+            AreaSupport::generateOutlineInsets(
+                support_infill_part.insets,
+                support_infill_part.outline,
+                wall_line_count,
+                support_line_width);
+            assert(support_infill_part.gradual_infill_areas_per_combine_per_density.size() == 0);
+
+            support_infill_part.inner_infill_areas.clear();
+            if (support_infill_part.insets.empty())
```

Comment:
If support gets more insets than one you can also continue if there are less insets than the required amount.

I think you should change this check to `support_infill_part.insets[skin_outline_count - 1].empty()`